includes/wikia/transaction/Transaction.php

```diff
@@ -237,15 +237,29 @@ public static function onAfterSmwfGetStore( $store ) {
 	 * - Cache-Control: public, max-age=2592000 (AssetsManager, cacheable)
 	 * - Cache-Control: private, must-revalidate, max-age=0 (special page, not cacheable)
 	 *
-	 * @param array $headers key - value list of HTTP response headers
+	 * @param array $headers list of headers to be sent as returned by headers_list()
 	 * @return bool|null will return null for maintenance / CLI scripts
 	 */
-	public static function isCacheable( $headers ) {
-		if ( empty( $headers['Cache-Control'] ) ) {
+	public static function isCacheable( array $headers ) {
+		/**
+		 * $headers will have entries like the following ones:
+		 *
+		 * 'X-Served-By: dev-macbre'
+		 * 'X-Trace-Id: 2d425caa-19a9-4f78-a5e7-8bdda3286f0d'
+		 *
+		 * Map it to an associative array
+		 */
+		$headersMap = [];
+		foreach($headers as $header) {
+			list($key, $val) = explode(':', $header, 2);
+			$headersMap[ strtoupper( $key ) ] = $val;
```

Comment:
I'd say it's more common to use `strtolower` in such cases but it's for sure arguable